src/PostgREST/Statements.hs

```diff
@@ -128,36 +128,39 @@ standardRow = (,,,,,) <$> nullableColumn HD.int8 <*> column HD.int8
 
 type ProcResults = (Maybe Int64, Int64, ByteString, Either SimpleError [GucHeader], Either SimpleError (Maybe Status))
 
-callProcStatement :: Bool -> SqlQuery -> SqlQuery -> SqlQuery -> Bool ->
+callProcStatement :: SqlQuery -> SqlQuery -> SqlQuery -> Bool ->
                      Bool -> Bool -> Bool -> Bool -> Maybe FieldName -> PgVersion ->
                      H.Statement ByteString ProcResults
-callProcStatement returnsScalar callProcQuery selectQuery countQuery countTotal isSingle asCsv asBinary multObjects binaryField pgVer =
+callProcStatement callProcQuery selectQuery countQuery countTotal isSingle asCsv asBinary multObjects binaryField pgVer =
   unicodeStatement sql (param HE.unknown) decodeProc True
   where
     sql = [qc|
-      WITH {sourceCTEName} AS ({callProcQuery})
+      WITH {sourceCTEName} AS ({callProcQuery}),
+      pgrst_select AS ({selectQuery}),
+      -- using to_json()::jsonb here, because pg9.4 does not have to_jsonb() yet
+      pgrst_to_json AS (SELECT to_json(pgrst_select)::jsonb AS pgrst_json FROM pgrst_select)
```

Comment:
Just using `to_jsonb` would be a lot better, probably also for performance.

We might consider dropping support for pg 9.4 - it's final release was in February and it's not supported anymore.

Reply:
It'd be better to drop pg 9.4 on another PR, I think there are changes we need to make on nix.

@monacoremo Perhaps you'd like to work on this?

Reply:
No, the tests are broken on purpose. I admit, my text above is quite lengthy.... so I'm quoting myself here:

> The last part changes the return values of RPCs in some ways that break existing tests. I left those tests unchanged for now, to have them shown up with expected and new output in the CI test output. 

Once we drop support for pg 9.4, we can change `to_json(pgrst_select)::jsonb` to `to_jsonb(pgrst_select)`. So instead of converting twice, we only need to do it once.

I have prepared a branch to drop 9.4 at any point soon. If you're ok with dropping it now, I can open a PR.